src/sync/sync_permission.hpp

```diff
@@ -150,11 +156,19 @@ class Permissions {
     using ConfigMaker = std::function<Realm::Config(std::shared_ptr<SyncUser>, std::string url)>;
 
     // Callback used to asynchronously vend a `PermissionResults` object.
+    // DEPRECATED. Will be replaced by RawPermissionResultsCallback in the future.
     using PermissionResultsCallback = std::function<void(std::unique_ptr<PermissionResults>, std::exception_ptr)>;
 
+    // Callback used to asynchronously vend permissions results.
+    using RawPermissionResultsCallback = std::function<void(Results, std::exception_ptr)>;
```

Comment:
Can you elaborate on why the new type alias / function is necessary given what this PR claims it is about?